specs/MultiProfile.md

```diff
@@ -165,6 +165,28 @@ void ScenarioCookieManagement::DeleteAllCookies()
     CHECK_FAILURE(m_cookieManager->DeleteAllCookies();
 }
 ```
+
+### Delete profile
+
+```cpp
+HRESULT AppWindow::DeleteProfile(ICoreWebView2Controller* controller)
+{
+    wil::com_ptr<ICoreWebView2> coreWebView2;
+    CHECK_FAILURE(controller->get_CoreWebView2(&coreWebView2));
+    auto webview7 = coreWebView2.try_query<ICoreWebView2_7>();
+    if (webview7)
+    {
+        wil::com_ptr<ICoreWebView2Profile> profile;
+        CHECK_FAILURE(webview7->get_Profile(&profile));
+        auto profile2 = profile.try_query<ICoreWebView2StagingProfile4>;
+        if (profile2)
+        {
+            CHECK_FAILURE(profile2->Delete());
+        }
```

Comment:
We should also show in the sample code how the caller deals with the results of calling Delete. The corewebview2 will be closed. Do they immediately switch to a new corewebview2 in a different profile and let the app continue using a webview2? Or the app is shutting down and deleting the profile but needs to wait for the browser process to exit before the app shuts down?

Reply:
The user can wait in the render process exit call back. The sample codes are in specs/ExtendedProcessFailed.md file, we just need to add a Enum reason entry: REASON_ENTRY(COREWEBVIEW2_PROCESS_FAILED_REASON_PROFILE_DELETED).

And the app will not exit, when the browser process exit (the delete API will close WV2, and shut down its render process, when all WV2 has exited, the browser process exit). So, the user can create another WV2 or shut down the app, or do nothing. It can be said like this, the user can treat the Delete API as an ordinary function, as long as not use Profile object they can do anything.

Reply:
Your documentation below says that the profile directory is deleted after the browser process exits. If the app is trying to delete the profile before the app exits and the app waits for all the ProcessFaileds to say the CoreWebView2s are closed and then it exits, can it exit before the profile folder is deleted?

Reply:
Yes, the app can shut down immediately after call Delete API.